addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/discovery/SqueezeBoxServerDiscoveryService.java

```diff
@@ -0,0 +1,275 @@
+			}
+		};
+
+		try {
+			// setup socket
+			startSocket();
+			thread.start();
+			receiveDiscoveryMessage();
+		} catch (Exception e) {
+			logger.debug("Problem scanning for Squeeze Server", e);
+		} finally {
+			// make sure we clean up.
+			try {
+				stopSocket();
+			} catch (Exception ignored) {
+
+			}
+		}
+	}
+
+	/**
+	 * Creates our UDP socket
+	 */
+	private void startSocket() throws Exception {
+
+		socket = new DatagramSocket();
+		socket.setBroadcast(true);
+		socket.setSoTimeout(10000);
+	}
+
+	/**
+	 * Stops and cleans up our UDP socket
+	 */
+	private void stopSocket() throws Exception {
+		if (socket != null)
+			socket.close();
+	}
+
+	/**
+	 * Based off the MaxCube discovery code, thanks! Sends discovery messages on
+	 * the network.
+	 * 
+	 * @param discoverString
+	 */
+	private void sendDiscoveryMessage() {
+		try {
+			byte[] sendData = discoPacket.getBytes();
+			// Broadcast the message over all the network interfaces
+			Enumeration<NetworkInterface> interfaces = NetworkInterface
+					.getNetworkInterfaces();
+			while (interfaces.hasMoreElements()) {
+				NetworkInterface networkInterface = (NetworkInterface) interfaces
+						.nextElement();
+				if (networkInterface.isLoopback() || !networkInterface.isUp()) {
+					continue;
+				}
+				for (InterfaceAddress interfaceAddress : networkInterface
+						.getInterfaceAddresses()) {
+					InetAddress[] broadcast = new InetAddress[3];
+					broadcast[0] = InetAddress.getByName("224.0.0.1");
+					broadcast[1] = InetAddress.getByName("255.255.255.255");
+					broadcast[2] = interfaceAddress.getBroadcast();
+					for (InetAddress bc : broadcast) {
+						if (bc != null) {
+							try {
+								logger.trace("Broadcasting {}", sendData);
+								DatagramPacket sendPacket = new DatagramPacket(
+										sendData, sendData.length, bc,
+										DISCO_PORT);
+								socket.send(sendPacket);
+							} catch (IOException e) {
+								logger.debug(
+										"IO error during SqueezeBoxServer discovery: {}",
+										e.getMessage());
+							} catch (Exception e) {
+								logger.debug(e.getMessage(), e);
+							}
+							logger.trace(
+									"Request packet sent to: {} Interface: {}",
+									bc.getHostAddress(),
+									networkInterface.getDisplayName());
+						}
+					}
+				}
+			}
+			logger.trace("Done looping over all network interfaces. Now waiting for a reply!");
+
+		} catch (IOException e) {
+			logger.debug("IO error during SqueezeBoxServer discovery: {}",
+					e.getMessage());
+		}
+	}
+
+	/**
+	 * Based off the MaxCube discovery code, thanks! Listens for replies to our
+	 * discovery message
+	 */
+	private void receiveDiscoveryMessage() {
+		try {
+			while (discoveryRunning) {
+				// Wait for a response
+				byte[] recvBuf = new byte[1500];
+				DatagramPacket receivePacket = new DatagramPacket(recvBuf,
+						recvBuf.length);
+				logger.trace("Waiting for packets");
+				socket.receive(receivePacket);
```

Comment:
Wlll this method block, if nothing is received (anymore)? How does the method terminate in this case when discoveryRunning is set to false?